src/VCR/Storage/AbstractStorage.php

```diff
@@ -50,12 +51,23 @@
     /**
      * Creates a new file store.
      *
+     * If the cassetteName contains PATH_SEPARATORs, subfolders of the
+     * cassettePath are autocreated when not existing.
+     *
      * @param string $cassettePath Path to the cassette directory.
-     * @param string $cassetteName Path to a file, will be created if not existing.
+     * @param string $cassetteName Path to the cassette file, relative to the path.
      */
     public function __construct($cassettePath, $cassetteName, $defaultContent = '[]')
     {
-        $file = $cassettePath . DIRECTORY_SEPARATOR . $cassetteName;
+        if (!is_dir($cassettePath)) {
+            throw new VCRException(sprintf('Cassette path "%s" is not existing or not a directory', $cassettePath), 0);
```

Comment:
oh, of course, sorry. i also suggest we keep it here - if people use a more IoC approach instead of going through the static factory, it will become useful. compared to the rest of the operations, a duplicate directory check should not be that expensive.